Machina/Sockets/RawCaptureSocket.cs

```diff
@@ -139,7 +139,19 @@ private void OnReceive(IAsyncResult ar)
                     _ = _socket.BeginReceive(_currentBuffer, 0, _currentBuffer.Length, SocketFlags.None, new AsyncCallback(OnReceive), null);
 
                     if (received > 0)
-                        _pendingBuffers.Enqueue(new Tuple<byte[], int>(buffer, received));
+                    {
+                        //Cut off useless buffer when is less than half of buffer size for saving memory
+                        if (BUFFER_SIZE / 2 > received)
```

Comment:
Can you write more about what the goal of this PR is?

The code you added here will be executed on every network receive, and will allocate a new byte array.   This increases memory allocations.  if the goal is to prevent extra bytes at the end of the buffer, a better way would be to not reallocate _currentBuffer in the lines above, and just copy the received bytes (assuming received>0) to a new array and enque that before calling BeginReceive().  But, I didn't do that because the code was previously pulling buffers from a fixed-length buffer pool, now removed due to other concerns.  And, that will introduce slightly more time where the socket kernel buffer will receive data before it is polled again.

Can you describe where you observed issues with 64kb buffers?

Reply:
Thanks for looking this PR and your suggestion! My goal is prevent extra unused bytes in buffer, in there it cost more cpu time for lesser space occupied in `_pendingbuffer`, so when there are so many small packet (e.g 50B packets), the memory it used won't growing repaidly. (The origin purpose is, in FFXIV process, ghub will create an connection to its agent for implementing its feature, and its packets will also captured by machina, and look like it using LOTS of packet)

Also I will optimize it according to your suggestion, and I will update the goal in description.

Reply:
Done, now is ready for reviewing.